### PR TITLE
DAOS-6138 test: Allow tearDown() 60s to complete

### PR DESCRIPTION
**Pull request by brianjmurrell**
Even if the test times out.  Currently if a test uses up the timeout,
tearDown() gets no time to run, leaving behind previous timed-out test
state and potentially causing the next test to fail.

Also add skipping pragma for the Leap 15 build.

Refactor ftest.sh to remove inline scripts.

Skip Unit Tests if centos7-gcc build stage was skipped since the
product from the centos7-gcc is used in the Unit Test stage.

Add some diagnostics on what "file" thinks a core file is when the
executuable name cannot be parsed out.

### CODE REVIEW

File: src/tests/ftest/scripts/main.sh [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+pushd "$PREFIX"/lib/daos/TESTING/ftest
+
+# make sure no lingering corefiles or junit files exist
+rm -f core.* ./*_results.xml
+
+# see if we just wanted to set up
+if ${SETUP_ONLY:-false}; then
+    exit 0
+fi
```
Comment (phender) [inline src/tests/ftest/scripts/main.sh:165]: As a future enhancement, I think:
1. this script should end before these lines (162-165)
2. these lines (162-165) should be moved back to `ftest.sh`
3. the remainder of this script (167+) should be moved to a new script (which would need to be called by `ftest.sh`. 

For manual testing users could then run the modified version of this script (up to line 161) once and then they could run either the new script (lines 167+) or their own launch.py command multiple times.

**Review by phender — APPROVED** [review]
LGTM

**Review by johannlombardi — DISMISSED** [review]

**Comment by brianjmurrell** [comment]
New failure: [DAOS-6319](https://jira.hpdd.intel.com/browse/DAOS-6319)

**Comment by brianjmurrell** [comment]
But ready for re-review.  Only change should be reverting accidental raft change.

**Pull request merged**